### PR TITLE
setting up prometheus

### PR DESCRIPTION
**Pull request by tanneberger**

### CODE REVIEW

**Comment by marenz2569** [comment]
- [ ] test prometheus exporter

**Comment by marenz2569** [comment]
- [ ] test prometheus exporter

File: include/prometheus_gauge_populator.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+#include <prometheus/gauge.h>
+
+namespace gr {
+namespace prometheus {
+
+/// This block takes a float as an input and writes in into a prometheus gauge
+class prometheus_gauge_populator : virtual public block {
+public:
+  using sptr = boost::shared_ptr<prometheus_gauge_populator>;
```
Comment (tanneberger) [inline include/prometheus_gauge_populator.h:15]: no just no lets use `std::share_ptr` or is there an actually convincing reason? 

or are we forced here to use a boost smart pointer to get the datafrom gnuradio?

File: include/prometheus_gauge_populator.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  prometheus_gauge_populator(::prometheus::Gauge& gauge);
+
+  static sptr make(::prometheus::Gauge& gauge);
```
Comment (tanneberger) [inline include/prometheus_gauge_populator.h:19]: trailing return types.

File: include/prometheus_gauge_populator.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  int general_work(int noutput_items, gr_vector_int& ninput_items, gr_vector_const_void_star& input_items,
+                   gr_vector_void_star& output_items) override;
+
+private:
+  /// the prometheus gauge we are populating with this block
+  ::prometheus::Gauge& gauge_;
```
Comment (tanneberger) [inline include/prometheus_gauge_populator.h:26]: `gauge_{}`

File: include/prometheus_gauge_populator.h (hunk 1 above)
Comment (marenz2569) [reply include/prometheus_gauge_populator.h]: we need to use what gnuradio does, sadly

**Review by marenz2569 — APPROVED** [review]

**Pull request merged**